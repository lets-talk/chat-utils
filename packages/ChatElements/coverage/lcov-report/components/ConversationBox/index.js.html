<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ConversationBox/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ConversationBox</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import moment from 'moment';
&nbsp;
// Other components used by this component
import Avatar from '../Avatar';
&nbsp;
// Component styles
import './index.scss';
&nbsp;
const ConversationBox = (props) =&gt; {
  const { conversation } = props;
  const tags = conversation.tags ? [conversation.tags[0]] : <span class="branch-1 cbranch-no" title="branch not covered" >[];</span>
  const lastMessageType = conversation.last_message.type ? <span class="branch-0 cbranch-no" title="branch not covered" >conversation.last_message.type </span>: '';
  const lastMessageReaded = conversation.last_message.readed;
  const timeAgo = moment(conversation.last_message.created_at).fromNow(true);
&nbsp;
  return (
    &lt;div
      role="button"
      tabIndex="-3"
      className={classNames(
        'letstalk-cbox', lastMessageType, props.className,
        {
          'not-read': !lastMessageReaded,
        }
      )}
      onClick={props.onClick}
      onKeyPress={props.onClick}
    &gt;
      {
        conversation.client.avatar &amp;&amp;
        &lt;div
          className={classNames('cbox-left', 'letstalk-cbox-avatar-container')}
        &gt;
          &lt;Avatar
            src={conversation.client.avatar}
            withStatus={false}
            size="small"
            status="online"
          /&gt;
        &lt;/div&gt;
      }
&nbsp;
      &lt;div
        className={classNames('cbox-right', 'cbox-row')}
      &gt;
        &lt;div className="cbox-column"&gt;
          &lt;div className="cbox-title"&gt;{conversation.client.name}&lt;/div&gt;
          &lt;div className="cbox-time-info-col"&gt;{!lastMessageReaded &amp;&amp; &lt;div className="cbox-circle"&gt;&amp;nbsp;&lt;/div&gt;}
            &lt;div className="cbox-time"&gt;{timeAgo}&lt;/div&gt;
          &lt;/div&gt;
        &lt;/div&gt;
&nbsp;
        &lt;div className="cbox-row"&gt;
          &lt;div className="cbox-column"&gt;
            &lt;div className="cbox-subtitle"&gt;{conversation.last_message.content}&lt;/div&gt;
            &lt;div className="cbox-tags"&gt;
              {tags.map((tag, index) =&gt; (
                &lt;span key={index} className="cbox-tag"&gt;
                  {tag.name}
                &lt;/span&gt;
              ))}
            &lt;/div&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
&nbsp;
    &lt;/div&gt;
  );
};
&nbsp;
ConversationBox.propTypes = {
  conversation: PropTypes.object.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
};
&nbsp;
ConversationBox.defaultProps = {
  onClick: null,
};
&nbsp;
&nbsp;
export default ConversationBox;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
