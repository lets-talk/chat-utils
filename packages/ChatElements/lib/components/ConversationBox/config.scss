@import "../../common/styles/variables";

$messageBox-background: $color-background;
$messageBox-color: $color-text;
$messageBox-agent-background: $color-background-secondary;
$messageBox-agent-color: $color-text-secondary;

$messageBox-font-size: $font-size-small;
$messageBox-title-font-size: $font-size-normal;
$messageBox-title-color: $color-white;
$messageBox-time-font-size: $font-size-tiny;

$messageBox-side-margin: 10px;
$messageBox-border-radius: 5px;
$messageBox-box-shadow: 0px rgba(0, 0, 0, .2);
