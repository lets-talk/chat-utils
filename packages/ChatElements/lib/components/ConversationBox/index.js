import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import moment from 'moment';

// Other components used by this component
import Avatar from '../Avatar';

// Component styles
import './index.scss';

const ConversationBox = (props) => {
  const { conversation } = props;

  return (
    <div
      role="button"
      tabIndex="-3"
      className={classNames('letstalk-container-cbox', props.className)}
      onClick={props.onClick}
      onKeyPress={props.onClick}
    >
      {
        conversation.client.avatar &&
        <div
          className={classNames('letstalk-cbox-avatar-container')}
        >
          <Avatar
            src={conversation.client.avatar}
            withStatus={false}
            size="xsmall"
            status="online"
          />
        </div>
      }

      <div
        className={classNames('letstalk-cbox')}
      >
        <div className="cbox-title-row">
          {conversation.client.name}
        </div>

        <div className="cbox-subtitle-row">
          {conversation.last_message.content}
        </div>
      </div>

    </div>
  );
};

ConversationBox.propTypes = {
  conversation: PropTypes.object.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
};

ConversationBox.defaultProps = {
  onClick: null,
};


export default ConversationBox;
