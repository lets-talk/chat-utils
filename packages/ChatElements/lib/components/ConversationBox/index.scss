@import './config';

.letstalk-container-cbox {
    flex-direction: column;
    display: block;
    overflow: hidden;
    min-width: 200px;
}

.letstalk-cbox {
  position: relative;
  background-color: $messageBox-background;
  color: $messageBox-color;
  border-radius: $messageBox-border-radius;
  box-shadow: $messageBox-box-shadow;
  margin-left: $messageBox-side-margin;
  margin-right: 5px;
  margin-top: 3px;
  flex-direction: column;
  margin-bottom: 3px;
  padding: 15px;
  float: left;
}

.letstalk-cbox-time {
  display: inline-block;
  width: 50px;
  text-align: right;
  font-size: $font-size-tiny;
  position: relative;
}
