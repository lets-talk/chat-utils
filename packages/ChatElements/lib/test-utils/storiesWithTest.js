import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
