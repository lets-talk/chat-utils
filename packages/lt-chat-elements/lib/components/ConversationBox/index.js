import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import styled from 'styled-components';

import { ellipsis, flexRow, flexColumn } from '../../utils/style';

// Other components used by this component
import TagList from '../TagList';
import Avatar from '../Avatar';

const bgColorChooser = (theme, type) => {
  if (type === 'internal') return theme.components.conversationBox.internalBackgroundColor;
  else if (type === 'important') return theme.components.conversationBox.importantBackgroundColor;

  return theme.palette.common.white;
};

const ConversationBoxWrapper = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  flex: 1;
  flex-wrap: nowrap;
  flex-direction: row;
  justify-content: flex-start;
  min-height: 50px;
  background-color: ${(props) => bgColorChooser(props.theme, props.type)};
`;

const CBoxRow = styled.div`
  ${flexRow()}
`;

const CBoxColumn = styled.div`
  ${flexColumn()}
`;

const CBoxRowLeft = styled.div`
  min-width: 50px;
  order: 1;
  padding: 10px;
`;

const CBoxRowRight = styled.div`
  ${flexRow()}
  position: relative;
  width: calc(100% - 80px);
  flex-basis: 1;
  order: 2;
  padding: 10px;
  padding-left: 0px;
`;

const TitleDiv = styled.div`
  flex: 1;
  order: 1;

  color: ${(props) => props.theme.typography.sizes.body.normal.dark.primary.color};
  font-size: ${(props) => props.theme.typography.sizes.body.normal.dark.primary.fontSize};
  font-weight: ${(props) => props.theme.typography.sizes.body.normal.dark.primary.fontWeight};
  ${ellipsis('100%')}
`;

const SubTitleDiv = styled.div`
  flex: 1;
  order: 1;

  color: ${(props) => props.theme.typography.sizes.footnote.normal.dark.secondary.color};
  font-size: ${(props) => props.theme.typography.sizes.footnote.normal.dark.secondary.fontSize};
  font-weight: ${(props) => props.theme.typography.sizes.footnote.normal.dark.secondary.fontWeight};
  ${ellipsis('100%')}
`;

const TimeInfoDiv = styled.div`
  display: flex;
  align-items: center;
  order: 2;

  color: ${(props) => props.theme.typography.sizes.caption.normal.dark.secondary.color};
  font-size: ${(props) => props.theme.typography.sizes.caption.normal.dark.secondary.fontSize};
  font-weight: ${(props) => props.theme.typography.sizes.caption.normal.dark.secondary.fontWeight};

  div {
    display: inline-block;
    text-align: right;
    font-size: ${(props) => props.theme.typography.sizes.small.normal.dark.fontSize};
    position: relative;
    color: ${(props) =>
    (props.readed && props.theme.palette.text.primary)
      || (!props.readed && props.theme.typography.sizes.caption.normal.accent.color)};
  }
`;

const CircleDiv = styled.div`
  width: 6px;
  height: 6px;
  border-radius: 50%;
  margin-right: 3px;
  background-color: ${(props) => props.theme.palette.background.primary};
`;

const ConversationBox = (props) => {
  const { conversation, onClickAction } = props;
  const tags = conversation.tags ? conversation.tags : [];
  const lastMessageType = conversation.last_message.type ? conversation.last_message.type : '';
  const lastMessageReaded = conversation.last_message.readed;
  const timeAgo = moment(conversation.last_message.created_at).fromNow(true);

  return (
    <ConversationBoxWrapper
      className={props.className}
      type={lastMessageType}
      onClick={onClickAction}
    >
      {
        conversation.client.avatar &&
        <CBoxRowLeft>
          <Avatar
            src={conversation.client.avatar}
            withStatus={false}
            size="small"
            status="online"
          />
        </CBoxRowLeft>
      }

      <CBoxRowRight>
        <CBoxColumn>
          <TitleDiv>{conversation.client.name}</TitleDiv>
          <TimeInfoDiv readed={lastMessageReaded}>{!lastMessageReaded && <CircleDiv>&nbsp;</CircleDiv>}
            <div>{timeAgo}</div>
          </TimeInfoDiv>
        </CBoxColumn>

        <CBoxRow>
          <CBoxColumn>
            <SubTitleDiv>{conversation.last_message.content}</SubTitleDiv>
            <TagList tags={tags} />
          </CBoxColumn>
        </CBoxRow>
      </CBoxRowRight>

    </ConversationBoxWrapper>
  );
};

ConversationBox.propTypes = {
  /**
   * The Conversation object
   */
  conversation: PropTypes.object.isRequired,
  /**
   * Function callback called when clicking on an item
   */
  onClickAction: PropTypes.func,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
};

ConversationBox.defaultProps = {
  onClickAction: null,
  className: 'LT-ConversationBox-Wrapper',
};

export default ConversationBox;
