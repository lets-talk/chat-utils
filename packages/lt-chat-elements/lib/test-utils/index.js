// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
