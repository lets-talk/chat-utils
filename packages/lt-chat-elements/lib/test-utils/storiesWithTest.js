import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
