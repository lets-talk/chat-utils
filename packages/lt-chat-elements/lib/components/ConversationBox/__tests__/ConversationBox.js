import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { renderWithTheme } from '../../../test-utils';
import ConversationBox from '../index';

describe('ConversationBox component', () => {
  // Fake the Date.now so test always returns 9 hours
  // This is equivalent to the 19/11/2017 -> 1511136000000
  Date.now = jest.fn(() => new Date(Date.UTC(2017, 10, 20)).valueOf());
  const conversation = {
    client: {
      name: 'Georgia Harmon',
      avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    },
    last_message: {
      type: 'normal',
      status: 'read',
      content: 'Lorem ipusum la la ejemplo largo',
      created_at: '2017-11-20T09:43:57.000-0300',
    },
    tags: [
      {
        name: 'SPU-CLAVE',
      },
    ],
  };

  it('should render with minimal props', () => {
    const component = shallow(<ConversationBox conversation={conversation} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      conversation,
      onClick: () => null,
    };

    const component = shallow(<ConversationBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have the correct styles (Those that depends on the theme) type=internal', () => {
    const props = {
      conversation: {
        ...conversation,
        last_message: {
          ...conversation.last_message,
          type: 'internal',
        },
      },
      onClick: () => null,
    };
    const tree = renderWithTheme(<ConversationBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', 'rgba(255,213,0,0.2)');
  });

  it('should have the correct styles (Those that depends on the theme) type=important', () => {
    const props = {
      conversation: {
        ...conversation,
        last_message: {
          ...conversation.last_message,
          type: 'important',
        },
      },
      onClick: () => null,
    };
    const tree = renderWithTheme(<ConversationBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', 'rgb(255,246,244)');
  });
});
