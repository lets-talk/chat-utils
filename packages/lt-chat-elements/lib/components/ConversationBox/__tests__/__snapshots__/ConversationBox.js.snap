// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationBox component should render with custom props 1`] = `
<div
  className="letstalk-cbox not-read"
>
  <div
    className="cbox-left letstalk-cbox-avatar-container"
  >
    <Avatar
      alt=""
      size="small"
      src="https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg"
      status="online"
      type="circle"
      withStatus={false}
    />
  </div>
  <div
    className="cbox-right cbox-row"
  >
    <div
      className="cbox-column"
    >
      <div
        className="cbox-title"
      >
        Georgia Harmon
      </div>
      <div
        className="cbox-time-info-col"
      >
        <div
          className="cbox-circle"
        >
           
        </div>
        <div
          className="cbox-time"
        >
          13 hours
        </div>
      </div>
    </div>
    <div
      className="cbox-row"
    >
      <div
        className="cbox-column"
      >
        <div
          className="cbox-subtitle"
        >
          Lorem ipusum la la ejemplo largo
        </div>
        <div
          className="cbox-tags"
        >
          <span
            className="cbox-tag"
            key="0"
          >
            SPU-CLAVE
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConversationBox component should render with minimal props 1`] = `
<div
  className="letstalk-cbox not-read"
>
  <div
    className="cbox-left letstalk-cbox-avatar-container"
  >
    <Avatar
      alt=""
      size="small"
      src="https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg"
      status="online"
      type="circle"
      withStatus={false}
    />
  </div>
  <div
    className="cbox-right cbox-row"
  >
    <div
      className="cbox-column"
    >
      <div
        className="cbox-title"
      >
        Georgia Harmon
      </div>
      <div
        className="cbox-time-info-col"
      >
        <div
          className="cbox-circle"
        >
           
        </div>
        <div
          className="cbox-time"
        >
          13 hours
        </div>
      </div>
    </div>
    <div
      className="cbox-row"
    >
      <div
        className="cbox-column"
      >
        <div
          className="cbox-subtitle"
        >
          Lorem ipusum la la ejemplo largo
        </div>
        <div
          className="cbox-tags"
        >
          <span
            className="cbox-tag"
            key="0"
          >
            SPU-CLAVE
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
